src/main/java/org/opentripplanner/routing/api/request/request/RouteViaRequest.java

```diff
@@ -1,16 +1,49 @@
 package org.opentripplanner.routing.api.request.request;
 
+import java.util.ArrayList;
 import java.util.List;
-
-//  User request: from/to time/location; preferences slack/cost/reluctance
+import org.opentripplanner.routing.api.request.RouteRequest;
+import org.opentripplanner.routing.api.request.preference.RoutingPreferences;
 
 // TODO VIA: Javadoc
 public class RouteViaRequest {
 
-  List<ViaLocation> viaPoints;
+  private final RoutingPreferences preferences;
+
+  private final RouteRequest routeRequest;
+
+  private final List<ViaLeg> viaLegs = new ArrayList<>();
+
+  public RouteViaRequest(
+    RoutingPreferences preferences,
+    List<ViaLocation> viaLocations,
+    List<JourneyRequest> viaJourneys,
+    RouteRequest routeRequest
+  ) {
+    this.preferences = preferences;
+    // Number of trips must match viaPoints + 1 and is limited to max 5 trips
+    this.routeRequest = routeRequest;
+
+    // TODO: Assert viaJourneys = viaLocations + 1
+
+    // Last ViaLeg has no ViaLocation
+    for (int i = 0; i < viaJourneys.size(); i++) {
+      var viaLocation = i < viaJourneys.size() - 1 ? viaLocations.get(i) : null;
+      viaLegs.add(new ViaLeg(viaJourneys.get(i), viaLocation));
+    }
+  }
+
+  public RoutingPreferences preferences() {
+    return this.preferences;
```

Comment:
Preferences is not used anywhere. Also it should probably not be a part of `RouteViaRequest`, but `ViaRoutingWorker` should probably take it as a constructor parameter, as it is the same for all parts of the journey.